scrapy/core/downloader/handlers/aiohttp.py

```diff
@@ -0,0 +1,52 @@
+"""Download handlers for http and https schemes"""
+import logging
+import asyncio
+
+import aiohttp
+from twisted.internet import defer
+
+from scrapy.http import Headers
+from scrapy.responsetypes import responsetypes
+
+logger = logging.getLogger(__name__)
+
+
+class HTTPDownloadHandler(object):
+
+    def __init__(self, settings):
+        self.settings = settings
+
+    def download_request(self, request, spider):
+        return _force_deferred(self._download_request(request, spider))
+
+    def _download_request(self, request, spider):
+        """Return a deferred for the HTTP download"""
+        headers=list((k.decode('latin1'), v.decode('latin1'))
+                     for k, vs in request.headers.items()
+                     for v in vs)
+
+        aioresponse = yield from aiohttp.request(
```

Comment:
> . Have no idea what do you mean under sending https urls over plain text.

Sorry, my statement is not clear at all. The idea is to use the same proxy protocol used for http:// urls but for https://, no _CONNECT_ method is send to establish a tunnel. The https:// url is send to the proxy in plain text, just like http urls are sent without tunneling.  

> 5- binding outgoing requests to specific source ips
> 
> Please elaborate. Do you mean binding client to specific network interface?

Twisted [allows us to bind](http://twistedmatrix.com/documents/15.3.0/api/twisted.internet.interfaces.IReactorTCP.connectTCP.html) outgoing connections to a specific source IP (and port, but not used). This is common practice for crawlers that run on a single servers with multiple assigned addresses. 

> 7- Degrade but do not fail to parse invalid responses
> 
> It sounds too common. We try to do some tricks for malformed headers but maybe it doesn't satisfy all your needs.

We have some specific bug reports related to this #345 and #210, invalid action lines, headers and differences in declared body size vs real body length are the most common examples seen.    

> Feel free to file missing issues into aiohttp bugtracker. Honestly I appreciate Pull Requests but even just tickets are also very useful for me.

If Scrapy goes for aiohttp for its download component, be cool you will get more pull requests than tickets, at least from its core developers.

> We use aiohttp client for our purposes but massive third party checker as Scrapy is can help with making aiohttp better. So even if you cannot or have no time for making aiohttp PR I'll gladly write it.

So far my impression is that aiohttp fits the bill for Scrapy, but I have to be sure others maintainers think the same because it is not just aiohttp, it is also depending on txtulip for a while and having a clear idea of what is the future of Twisted for Scrapy once we delegate the ioloop to asyncio. 

looking forward to collaborate together. thanks!